crates/project_model/src/lib.rs

```diff
@@ -27,14 +28,27 @@ pub use crate::{
 
 pub use proc_macro_api::ProcMacroClient;
 
-#[derive(Debug, Clone, Eq, PartialEq)]
+#[derive(Clone, Eq, PartialEq)]
 pub enum ProjectWorkspace {
     /// Project workspace was discovered by running `cargo metadata` and `rustc --print sysroot`.
     Cargo { cargo: CargoWorkspace, sysroot: Sysroot },
     /// Project workspace was manually specified using a `rust-project.json` file.
     Json { project: ProjectJson },
 }
 
+impl fmt::Debug for ProjectWorkspace {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ProjectWorkspace::Cargo { cargo, .. } => {
+                f.debug_struct("Cargo").field("n_packages", &cargo.packages().len()).finish()
```

Comment:
no info about sysroot?

Reply:
Not yet -- I plan to refactor sysroot handing (apparently, folks using project.json are not excited about specifing sysroot manually), so I'll fix the logging then. i hope :)